lib/hooks/useSchematicGroupsOverlay.ts

```diff
@@ -0,0 +1,217 @@
+import { useEffect } from "react"
+import { su } from "@tscircuit/soup-util"
+import type { CircuitJson } from "circuit-json"
+
+export const useSchematicGroupsOverlay = (
+  svgDivRef: React.RefObject<HTMLDivElement | null>,
+  circuitJson: CircuitJson,
+  circuitJsonKey: string,
+  showGroups: boolean,
+) => {
+  useEffect(() => {
+    if (
+      !svgDivRef.current ||
+      !showGroups ||
+      !circuitJson ||
+      circuitJson.length === 0
+    ) {
+      // Remove any existing group overlays
+      if (svgDivRef.current) {
+        const existingOverlays = svgDivRef.current.querySelectorAll(
+          ".schematic-group-overlay",
+        )
+        existingOverlays.forEach((overlay) => overlay.remove())
+      }
+      return
+    }
+
+    const svg = svgDivRef.current.querySelector("svg")
+    if (!svg) {
+      return
+    }
+
+    // Remove existing overlays first
+    const existingOverlays = svg.querySelectorAll(".schematic-group-overlay")
+    existingOverlays.forEach((overlay) => overlay.remove())
+
+    try {
+      // Get explicit groups first
+      const sourceGroups = su(circuitJson).source_group?.list() || []
+      const schematicComponents =
+        su(circuitJson).schematic_component?.list() || []
+
+      let groupsToRender: Array<{
+        id: string
+        name: string
+        components: any[]
+        color: string
+      }> = []
+
+      // Check if we have meaningful explicit groups (not just auto-generated default groups)
+      const hasMeaningfulGroups = sourceGroups.length > 0 && 
+        sourceGroups.some(group => group.name && group.name !== "default" && group.name !== "")
```

Comment:
check in prod, this condition doesn't make sense. Groups are never given the name default OR an empty string, they are given a generated name, you can just remove this condition